files/scripts/supervisor-proc-exit-listener

```diff
@@ -1,17 +1,30 @@
 #!/usr/bin/env python
 
+import getopt
 import os
 import signal
 import sys
 import syslog
 
+import swsssdk
+
 from supervisor import childutils
 
 # Contents of file should be the names of critical processes (as defined in
 # supervisor.conf file), one per line
 CRITICAL_PROCESSES_FILE = '/etc/supervisor/critical_processes'
 
-def main():
+# This table in databse contains the features for container and each
+# feature for a row will be configured a state or number.
+CONTAINER_FEATURE_TABLE_NAME = 'CONTAINER_FEATURE'
+
+def main(argv):
+    container_name = ''
+    opts, args = getopt.getopt(argv, "c:", ["container-name="])
+    for opt, arg in opts:
+        if opt in ("-c", "--container-name"):
+            container_name = arg
```

Comment:
What happens if the container name is not provided? What is the intended behavior?

Reply:
At here, we may have another corner case: the provided container name is invalid or wrong name. For this case, the auto-restart feature will be disabled. If container name is not provided, it will raise an exception. Obviously this needed to be fixed. If the container name is not provided, in the handler of exception, can we set it to be empty string to disable the auto-restart feature by default?

Reply:
I think it makes sense to disable to feature if the container name is missing or invalid/wrong. However, I think we should also log a message to syslog in these scenarios, since the script runs in the background, a syslog message will help inform the user that something is wrong. For instance, here I think we can do the following:

```
    container_name = None
    opts, args = getopt.getopt(argv, "c:", ["container-name="])
    for opt, arg in opts:
        if opt in ("-c", "--container-name"):
            container_name = arg

    if not container_name:
        syslog.syslog(syslog.LOG_ERROR, "Container name not specified. Exiting...")
        sys.exit(0)
```

I'm not sure if we should exit with zer or non-zero here, as we have configured supervisor to auto-restart the exit listener upon unexpected exit. I'm not sure what this behavior is like with exit listeners. I don't want it to get into an infinite loop of restarting the listener. Maybe you can test this behavior before choosing the exit code?

Reply:
Will test it.

Reply:
@jleveque I have tested the sys.exit(0) if the container name is not provided. If we call sys.exit(0), then event listener will exit with code 0. The supervisord will capture this event and
restart event listener. Then supervisord will capture this event again. Currently, supervisord will capture this event only 3 times since the default value for **startretries** set by supervisord is 3 (see this http://supervisord.org/configuration.html#program-x-section-settings). After that, the process of event listener will be put in **FATAL** state and never run again unless we manually use supervisorctl to start it.

Reply:
@jleveque I accidently deleted the comment and will write it again. I have tested using the sys.exit(0) and sys.exit(1) in the event listener if the container name is not provided. If this situation occurred, the event listener will exit with code 0 or non-zero, then supervisord will capture this event and restart event listener. After that, supervisord will capture this event again. Currently, supervisord will only restart the event listener 3 times since the value of **startretries** is set 3 by default (see this http://supervisord.org/configuration.html#program-x-section-settings). Finally the process of event listener will be set to be **FATAL** state and never run again unless we manually start it using the superviorctl to do it.

Reply:
OK, thanks. Then I think we should use non-zero exit codes, since it indicates an abnormal exit.